eppo_core/src/events/event_dispatcher.rs

```diff
@@ -0,0 +1,120 @@
+use crate::events::batch_event_processor::BatchEventProcessor;
+use crate::events::event::Event;
+use log::info;
+use std::sync::{Arc, Mutex};
+use tokio::sync::{Notify, OnceCell};
+use tokio::time::{interval, interval_at, Duration, Instant};
+
+#[derive(Debug, Clone)]
+pub struct EventDispatcherConfig {
+    pub sdk_key: String,
+    pub ingestion_url: String,
+    pub delivery_interval_ms: u64,
+    pub retry_interval_ms: u64,
+    pub max_retry_delay_ms: u64,
+    pub max_retries: Option<u32>,
+}
+
+pub struct EventDispatcher {
+    config: EventDispatcherConfig,
+    batch_processor: BatchEventProcessor,
+    delivery_task_active: Arc<Mutex<bool>>,
```

Comment:
I believe `Mutex<bool>` is blocking to the calling thread.

Depending on our need consider `Arc<AtomicBool>` or `tokio::sync::Mutex` - I will do some more reading to further understand the difference but the former is used in the library already.